controllers/util/provider.go

```diff
@@ -19,15 +19,20 @@ import (
 
 // GenerateOnboardingToken generates a token valid for a duration of "tokenLifetimeInHours".
 // The token content is predefined and signed by the private key which'll be read from supplied "privateKeyPath".
-func GenerateOnboardingToken(tokenLifetimeInHours int, privateKeyPath string) (string, error) {
+// The storageQuotaInGiB is optional, and it is used to limit the storage of PVC in the application cluster.
+func GenerateOnboardingToken(tokenLifetimeInHours int, privateKeyPath string, storageQuotaInGiB *int) (string, error) {
 	tokenExpirationDate := time.Now().
 		Add(time.Duration(tokenLifetimeInHours) * time.Hour).
 		Unix()
 
-	payload, err := json.Marshal(services.OnboardingTicket{
+	ticket := services.OnboardingTicket{
 		ID:             uuid.New().String(),
 		ExpirationDate: tokenExpirationDate,
-	})
+	}
+	if storageQuotaInGiB != nil {
+		ticket.StorageQuotaInGiB = *storageQuotaInGiB
```

Comment:
I think in the absence of body, this will always be Zero, is that fine?
```
var val int // gets a zero value
gen(_, _, ptr.To(val)) // you are always passing a pointer and so the 3rd param to generate isn't nil ever
```

I believe below could work, pls cross check
```
var val *int
t := unit * q.V
val = &t
gen(_, _, val)
```

Reply:
Yes, in the absence of a body, this will always be Zero.

You are correct. During my testing, it worked because after marshaling, the field was not being passed due to the omitempty tag being added at the struct level. I will make the changes as you suggested because that makes more sense to me.

Reply:
> due to the omitempty tag being added at the struct level
- ya, that could be fine as well and so I don't mind change this then unless other reviewers also want to change this.

Reply:
Please consider we do need to support nil in cases where a quota is not provided. We already defined that a quota with value of 0 (or negative value) is invalid and that a omitted quota should be nil

Reply:
Verified with 0 as well nil it working as expected.